met-web/src/components/engagement/admin/create/authoring/AuthoringContext.tsx

```diff
@@ -26,55 +27,67 @@ export const AuthoringContext = () => {
     const fetcher = useFetcher();
     const locationArray = window.location.href.split('/');
     const slug = locationArray[locationArray.length - 1];
+    const defaultDateValue = dayjs(new Date(1970, 0, 1));
     const engagementUpdateForm = useForm<EngagementUpdateData>({
         defaultValues: {
             id: 0,
-            name: '',
-            start_date: undefined,
-            end_date: undefined,
             status_id: 0,
+            taxon_id: 0,
+            content_id: 0,
+            name: '',
+            start_date: defaultDateValue,
+            end_date: defaultDateValue,
             description: '',
             rich_description: '',
             banner_filename: '',
             status_block: [],
             title: '',
             icon_name: '',
             metadata_value: '',
-            taxon_id: 0,
-            send_report: false,
+            send_report: undefined,
             slug: '',
             request_type: '',
         },
         mode: 'onSubmit',
         reValidateMode: 'onChange',
     });
-
     const onSubmit = async (data: EngagementUpdateData) => {
         fetcher.submit(
             createSearchParams({
-                id: data.id.toString(),
+                id: 0 === data.id ? '' : data.id.toString(),
+                status_id: 0 === data.status_id ? '' : data.status_id.toString(),
+                taxon_id: 0 === data.taxon_id ? '' : data.taxon_id.toString(),
+                content_id: 0 === data.content_id ? '' : data.content_id.toString(),
                 name: data.name,
-                start_date: data.start_date.format('YYYY-MM-DD'),
-                end_date: data.end_date.format('YYYY-MM-DD'),
+                start_date:
+                    '1970-01-01' === data.start_date.format('YYYY-MM-DD') ? '' : data.start_date.format('YYYY-MM-DD'),
+                end_date:
+                    '1970-01-01' === data.start_date.format('YYYY-MM-DD') ? '' : data.end_date.format('YYYY-MM-DD'),
                 description: data.description,
                 rich_description: data.rich_description,
                 banner_filename: data.banner_filename,
                 status_block: data.status_block,
                 title: data.title,
                 icon_name: data.icon_name,
                 metadata_value: data.metadata_value,
-                taxon_id: data.taxon_id.toString(),
-                send_report: data.send_report ? 'true' : 'false',
+                send_report: getSendReportValue(data.send_report),
                 slug: data.slug,
                 request_type: data.request_type,
             }),
             {
                 method: 'post',
-                action: `/engagements/${data.id}/authoring/${slug}`,
+                action: `/engagements/${data.id}/details/authoring/${slug}`,
             },
         );
     };
 
+    const getSendReportValue = (valueToInterpret: boolean) => {
```

Comment:
So the data in createSearchParams all needs to be converted to strings. In this particular case, I just needed to convert a boolean value to a string equivalent as 'true' or 'false'. It's quite possible that there is another utility function that does this, I just don't know about it.

Reply:
Booleans can't be used with truthy checks because they can return false if their value is false, even though I want that conditional to pass on a false value. So I need to check if the value is undefined separately, then give a value that is either 'true' or 'false'.

Reply:
Yeah I was just curious how this operation for creating an HTTP request gets handled elsewhere and if we could use that. No worries if you want to just stick with this function you've created